### PR TITLE
Security utils more

### PR DESCRIPTION
**Pull request by millevlada**

### CODE REVIEW

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
 	static private final SecureRandom RANDOM = new SecureRandom();
 
-	public static final String ENCRYPTION_CIPHER = "Blowfish";
+	private enum Algorythm { Blowfish, AES }
```
Comment (fjalvingh) [inline common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: Algorithm

File: common/to.etc.alg/pom.xml [hunk 2]
```diff
@@ -51,7 +51,10 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
         </dependency>
-
+        <dependency>
```
Comment (fjalvingh) [inline common/to.etc.alg/pom.xml]: We do not need this.

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java [hunk 3]
```diff
@@ -283,29 +283,49 @@ static public boolean checkPassword(@NonNull String encodedPassword, @NonNull St
...
+	}
+
+	static public String decryptStringAes(String input, String password) throws Exception {
+		byte[] bytes = StringTool.decodeBase64(input);
+		return decryptStringFromBytes(bytes, password, Algorythm.AES);
+	}
+
+	static public String encryptStringAesHex(String input, String password) throws Exception {
+		byte[] bytes = encryptStringToBytes(input, password, Algorythm.AES);
+		return Hex.encodeHexString(bytes);
```
Comment (fjalvingh) [inline common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: StringTool.toHex()

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java [hunk 4]
```diff
@@ -283,29 +283,49 @@ static public boolean checkPassword(@NonNull String encodedPassword, @NonNull St
...
+	}
+
+	static public String decryptStringAesHex(String input, String password) throws Exception {
+		byte[] bytes = Hex.decodeHex(input.toCharArray());
```
Comment (fjalvingh) [inline common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: StringTool.fromHex

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java [hunk 5]
```diff
@@ -283,29 +283,49 @@ static public boolean checkPassword(@NonNull String encodedPassword, @NonNull St
...
 		return StringTool.encodeBase64ToString(bytes);
 	}
 
 	static public String decryptString(String input, String password) throws Exception {
 		byte[] bytes = StringTool.decodeBase64(input);
-		return decryptStringFromBytes(bytes, password);
+		return decryptStringFromBytes(bytes, password, Algorythm.Blowfish);
+	}
+
+	static public String encryptStringAes(String input, String password) throws Exception {
```
Comment (fjalvingh) [inline common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: Why not just have base64 and hex variants and pass in the algorithm? We now have to add a whole set of methods for every algorithm added...

**Review by fjalvingh — CHANGES_REQUESTED** [review]

File: common/to.etc.alg/pom.xml (hunk 2 above)
Comment (millevlada) [reply common/to.etc.alg/pom.xml]: ? It has handy methods that I use to encode to hex and to decode from hex. Why we should not use it?

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java (hunk 5 above)
Comment (millevlada) [reply common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: we had encryptString and decryptString in old code base that used Blowfish and base64. So, I had to keep it for backward compatibility. That is why I just added 2 more pairs of 2 lines methods, that I used in my code, with reusing the stuff that was there already. Isn't it better to have dedicated methods then combining the parameters in each call? I mean, can be easier to make the mistake.

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java (hunk 3 above)
Comment (millevlada) [reply common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: ok, I see, that is why we do not need that library... Ok, I'll switch to that.

File: common/to.etc.alg/pom.xml (hunk 2 above)
Comment (millevlada) [reply common/to.etc.alg/pom.xml]: ok, I see we have that StringTool. I just saw that Hex util used in other code, but I agree we do not need to introduce it here. Fixed.

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java (hunk 1 above)
Comment (millevlada) [reply common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: fixed

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java (hunk 5 above)
Comment (millevlada) [reply common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: ok, I made it as asked.

File: common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java (hunk 4 above)
Comment (millevlada) [reply common/to.etc.alg/src/main/java/to/etc/util/SecurityUtils.java]: fixed

**Review by fjalvingh — APPROVED** [review]

**Pull request merged**